docs/StandardLibraryProgrammersManual.md

```diff
@@ -125,6 +125,35 @@ Use of this attribute imposes limitations on what can be in the body. For more d
 
 This is currently used in the standard library as an additional annotation applied to @objc protocols signifying that any objects which conform to this protocol are not tagged. This means that (on Darwin platforms) such references, unlike AnyObject, have spare bits available from things like restricted memory spaces or alignment.
 
+#### `@_silgen_name`
+
+This attribute specifies the name that a declaration will have at link time. It is used for two purposes, the second of which is currently considered bad practice and should be replaced with shims:
+
+1. To specify the symbol name of a Swift function so that it can be called from Swift-aware C. Such functions have bodies.
+2. To provide a Swift declaration which really represents a C declaration. Such functions do not have bodies.
+
+##### Using `@_silgen_name` to call Swift from Swift-aware C
+
+Rather than hard-code Swift mangling into C code, `@_silgen_name` is used to provide a stable and known symbol name for linking. Note that C code still must understand and use the Swift calling convention (available in swift-clang) for such Swift functions (if they use Swift's CC). Example:
+
+```swift
+@_silgen_name("_swift_stdlib_destroyTLS")
+internal func _destroyTLS(_ ptr: UnsafeMutableRawPointer?) {
+  // ... implementation ...
+}
+```
+
+```C++
+__attribute__((swiftcall)) extern "C" void _swift_stdlib_destroyTLS(void *);
```

Comment:
Please use the reserved spelling for the calling convention.  `__attribute__((__swiftcall__))`.